tests/unit/test.config.js

```diff
@@ -0,0 +1,187 @@
+          },
+        },
+      });
+      const configObject = {
+        sourceDir: '/configured/source/dir',
+      };
+      const newArgv = applyConfigToArgv({argv, configObject, defaultValues});
+      assert.strictEqual(newArgv.sourceDir, configObject.sourceDir);
+    });
+
+    it('preserves a string value on the command line over all others', () => {
+      const cmdLineSrcDir = '/user/specified/source/dir/';
+      const {argv, defaultValues} = makeArgv({
+        userCmd: ['fakecommand', '--sourceDir', cmdLineSrcDir],
+        globalOpt: {
+          'source-dir': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'default/value/option/definition',
+          },
+        },
+      });
+      const configObject = {
+        sourceDir: '/configured/source/dir',
+      };
+      const newArgv = applyConfigToArgv({argv, configObject, defaultValues});
+      assert.strictEqual(newArgv.sourceDir, cmdLineSrcDir);
+    });
+
+    it('preserves default value of option if not in config', () => {
+      const {argv, defaultValues} = makeArgv({
+        userCmd: ['fakecommand'],
+        globalOpt: {
+          'source-dir': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'default/value/option/definition',
+          },
+        },
+      });
+      const configObject = {
+        foo: '/configured/foo',
+      };
+      const newArgv = applyConfigToArgv({argv, configObject, defaultValues});
+      assert.strictEqual(newArgv.sourceDir, 'default/value/option/definition');
+    });
+
+    it('preserves value on the command line if not in config', () => {
+      const cmdLineSrcDir = '/user/specified/source/dir/';
+      const {argv, defaultValues} = makeArgv({
+        userCmd: ['fakecommand', '--sourceDir', cmdLineSrcDir],
+        globalOpt: {
+          'source-dir': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'default/value/option/definition',
+          },
+        },
+      });
+      const configObject = {
+        foo: '/configured/foo',
+      };
+      const newArgv = applyConfigToArgv({argv, configObject, defaultValues});
+      assert.strictEqual(newArgv.sourceDir, cmdLineSrcDir);
+    });
+  });
+
+  describe('loadJSConfigFile', () => {
+    it('throws an error if the config file does not exist', () => {
+      return withTempDir (
+        (tmpDir) => {
+          assert.throws(() => {
+            loadJSConfigFile(tmpDir.path());
+          }, UsageError, /Cannot read config file/);
+        });
+    });
+
+    it('throws an error if the config file has syntax errors', () => {
+      return withTempDir (
+        (tmpDir) => {
+          const configFilePath = path.join(tmpDir.path(), 'config.js');
+          fs.writeFileSync(configFilePath,
+            `module.exports = {
+                sourceDir 'path/to/fake/source/dir',
```

Comment:
since this is only missing one character it's hard to read this code and immediately see that there is a syntax error in it. Can you either rewrite it to look like a more obvious syntax error or put a comment at the top explaining what the syntax error is?